pkg/ts/server.go

```diff
@@ -22,30 +22,57 @@ import (
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 
+	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/ts/tspb"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/stop"
 	gwruntime "github.com/grpc-ecosystem/grpc-gateway/runtime"
 )
 
 const (
 	// URLPrefix is the prefix for all time series endpoints hosted by the
 	// server.
 	URLPrefix = "/ts/"
+	// QueryWorkerMax determines the maximum number of worker goroutines that
+	// the time series server can use to service incoming queries.
+	QueryWorkerMax = 250
 )
 
+// TimeSeriesTestingKnobs provides a means for tests to override settings in the
+// test server that are not normally overridden.
+type TimeSeriesTestingKnobs struct {
+	// Adjusts the maximum number of query workers used by the server.
+	QueryWorkerMax *int
```

Comment:
Making this a pointer is slightly awkward. Elsewhere we use `0` as an indicator that a value should assume the default. For example, see `StoreConfig.SetDefaults`.